examples/weight_compression/openvino/tiny_llama/main.py

```diff
@@ -0,0 +1,211 @@
+import openvino as ov
+from datasets import load_dataset
+from optimum.intel import OVModelForCausalLM
+from transformers import AutoTokenizer
+from whowhatbench import Evaluator
+
+import nncf
+from nncf.common.logging import nncf_logger
+
+DataItem = TypeVar("DataItem")
+ModelInput = TypeVar("ModelInput")
+
+ROOT = Path(__file__).parent.resolve()
+MODEL_PATH = ROOT / "compressed_model.xml"
+
+COMPRESSION_MODE = nncf.parameters.CompressWeightsMode.INT4_SYM
+MAX_DROP = 0.2
+MIN_GROUP_SIZE = 64
+MAX_GROUP_SIZE = 128
+MIN_RATIO = 0.5
+MAX_RATIO = 1.0
+
+
+def compress_model(ov_model: ov.Model, nncf_dataset: nncf.Dataset, ratio: float, group_size: int) -> ov.Model:
+    """
+    Compress the given OpenVINO model using NNCF weight compression.
+
+    :param ov_model: The original OpenVINO model to be compressed.
+    :param nncf_dataset: A representative dataset for the weight compression algorithm.
+    :param ratio: The ratio between baseline and backup precisions
+    :param group_size: Number of weights (e.g. 128) in the channel dimension
+        that share quantization parameters (scale).
+    :return: The OpenVINO model with compressed weights.
+    """
+    optimized_ov_model = nncf.compress_weights(
+        ov_model.clone(), # we should clone the model because `compress_weights` works in place
+        dataset=nncf_dataset,
+        mode=COMPRESSION_MODE,
+        ratio=ratio,
+        group_size=group_size,
+        sensitivity_metric=nncf.parameters.SensitivityMetric.HESSIAN_INPUT_ACTIVATION,
+    )
+    return optimized_ov_model
+
+
+def evaluate_model(
+    hf_model: OVModelForCausalLM, original_ov_model: ov.Model, optimized_model: ov.Model, evaluator: Evaluator
+) -> float:
+    """
+    Get similarity of text generation between original and optimized models.
+
+    :param hf_model: The OpenVINO model for causal language modeling.
+    :param original_ov_model: The original OpenVINO model
+    :param optimized_model: The OpenVINO model with compressed weights.
+    :param evaluator: The evaluator object from whowhatbench Benchmark.
+    :return: The similarity score between the original and optimized models.
+    """
+    hf_model.model = optimized_model
+    hf_model.request = None
+    _, all_metrics = evaluator.score(hf_model)
+    hf_model.model = original_ov_model.clone()
+    hf_model.request = None
+    similarity = all_metrics["similarity"][0]
+    group_size = optimized_model.get_rt_info()['nncf']['weight_compression']['group_size'].value
+    ratio = optimized_model.get_rt_info()['nncf']['weight_compression']['ratio'].value
+    nncf_logger.info(
+        f"The similarity of model compressed with group_size={group_size}, ratio={ratio} is {similarity:.3f}"
+    )
+    return similarity
+
+
+def get_nncf_dataset(
+    data_source: Iterable[DataItem], transform_func: Optional[Callable[[DataItem], ModelInput]] = None
+) -> nncf.Dataset:
+    """
+    Create an NNCF dataset for the weight compression algorithm.
```

Comment:
Just for the future. It is good that you create a description for each function but this is not mandatory for the sample. Without description/typehints, it can make it smaller and even easier to digest.